test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -79,167 +87,320 @@ func (s *snapshottableTestSuite) SkipRedundantSuite(driver TestDriver, pattern t
+	framework.ExpectNoError(err)
+
+	// Get the bound snapshotContent
+	snapshotStatus := l.vs.Object["status"].(map[string]interface{})
+	snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+	l.vscontent, err = l.dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	snapshotContentSpec := l.vscontent.Object["spec"].(map[string]interface{})
+	volumeSnapshotRef := snapshotContentSpec["volumeSnapshotRef"].(map[string]interface{})
+
+	// Check SnapshotContent properties
+	ginkgo.By("checking the SnapshotContent")
+	framework.ExpectEqual(snapshotContentSpec["volumeSnapshotClassName"], l.vsc.GetName())
+	framework.ExpectEqual(volumeSnapshotRef["name"], l.vs.GetName())
+	framework.ExpectEqual(volumeSnapshotRef["namespace"], l.vs.GetNamespace())
+}
+
+// TestSnapshotDeleted tests the results of deleting a VolumeSnapshot
+// depending on the deletion policy currently set.
+func TestSnapshotDeleted(l *snapshottableLocal, sct SnapshotClassTest) {
+	var err error
+	snapshotStatus := l.vs.Object["status"].(map[string]interface{})
+	snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+
+	ginkgo.By("deleting the snapshot")
+	err = l.dc.Resource(SnapshotGVR).Namespace(l.vs.GetNamespace()).Delete(context.TODO(), l.vs.GetName(), metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		framework.Failf("Error deleting snapshot %s in namespace %s. Error: %v", l.vs.GetName(), l.vs.GetNamespace(), err)
+	}
+
+	ginkgo.By("checking the Snapshot has been deleted")
+	err = WaitForGVRDeletion(l.dc, SnapshotGVR, l.sc.Name, framework.Poll, framework.SnapshotDeleteTimeout)
+	framework.ExpectNoError(err)
+
+	if sct.DeletionPolicy == "Delete" {
+		ginkgo.By("checking the SnapshotContent has been deleted")
+		err = WaitForGVRDeletion(l.dc, SnapshotContentGVR, snapshotContentName, framework.Poll, framework.SnapshotDeleteTimeout)
+		framework.ExpectNoError(err)
+	} else if sct.DeletionPolicy == "Retain" {
+		ginkgo.By("checking the SnapshotContent has not been deleted")
+		err = WaitForGVRDeletion(l.dc, SnapshotContentGVR, snapshotContentName, 1*time.Second, 10*time.Second)
+		framework.ExpectError(err) // should fail deletion check
+
+		// prevent physical snapshotContent leaks
+		defer func() {
+			ginkgo.By("get the latest copy of volume snapshot content")
+			snapshotContent, err := l.dc.Resource(SnapshotContentGVR).Get(context.TODO(), l.vscontent.GetName(), metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			ginkgo.By("updating the the SnapshotContent to have Delete Deletion policy")
+			snapshotContent.Object["spec"].(map[string]interface{})["deletionPolicy"] = "Delete"
+			l.vsc, err = l.dc.Resource(SnapshotContentGVR).Update(context.TODO(), snapshotContent, metav1.UpdateOptions{})
+			framework.ExpectNoError(err)
```

Comment:
Can you please explain what are we trying to test here by updating the deletion policy? Since we are anyway manually deleting snapshot content object in next line. Does it mean that unless we update the deletion policy to "delete" for the snapshot content object, we cannot manually delete it too?

Reply:
Yes, that is correct. It's easier to update the snapshot content to have the deletion policy. Earlier in this PR, I was exec-ing into the container to check on this. This was causing problems (for differing test infra).  

I've added a comment to explain why this block is needed.

Reply:
Ack